src/Storages/Hive/StorageHive.cpp

```diff
@@ -705,6 +711,62 @@ Pipe StorageHive::read(
     return Pipe::unitePipes(std::move(pipes));
 }
 
+HiveFiles StorageHive::collectHiveFiles(
+    unsigned max_threads,
+    const SelectQueryInfo & query_info,
+    const HiveTableMetadataPtr & hive_table_metadata,
+    const HDFSFSPtr & fs,
+    const ContextPtr & context_,
+    PruneLevel prune_level) const
+{
+    std::vector<Apache::Hadoop::Hive::Partition> partitions = hive_table_metadata->getPartitions();
+    /// Hive table have no partition
+    if (!partition_name_types.empty() && partitions.empty())
+        return {};
+
+    /// Hive files to collect
+    HiveFiles hive_files;
+    /// Mutext to protect hive_files, which maybe appended in multiple threads
+    std::mutex hive_files_mutex;
+    ThreadPool pool{max_threads};
```

Comment:
If we will process every file/partition sequentially, will it really slow down query, so is it bottleneck?

Reply:
Yes. Because load minmax index will access HDFS, its latency will sum up when file/partition is processed sequentially